feat: [PL-47826]: Update delegate resource threshold env documentation (#5593)
harness/developer-hub: docs/platform/delegates/manage-delegates/auto-scale-using-replicas.md

     enabled: false
     minReplicas: 1
     maxReplicas: 10
     targetMemoryUtilizationPercentage: 80
     # targetCPUUtilizationPercentage: 80
   ```

2. Set `enabled` to `true`.

3. Specify the minimum and maximum number of replicas you want to use in the `minReplicas` and `maxReplicas` parameters.

   To fine-tune your autoscaling, you can set the `targetMemoryUtilizationPercentage` to add a new replica if Memory utilization exceeds this percentage

> 4. (Optional) Set the `targetCPUUtilizationPercentage` to add a new replica if CPU utilization exceeds this percentage.

Reviewer: ```suggestion
4. (Optional) Set the `targetCPUUtilizationPercentage` to add a new replica when CPU utilization exceeds this percentage.
```

5. Save the file, and restart your pods.

   When you create a deployment, Harness automatically spins up new replicas of your delegate as needed to ensure the deployment is completed.

> :::info CPU based HPA is not recommended as CPU assignments greater than 100% are usual and should not be a reason to scale or reject tasks. It should be consider when the CPU usage is above 100% for a long time. Harness recommends using Memory based HPA for autoscaling.

Reviewer: ```suggestion
:::info 
Using CPU-based HPA is not advisable as CPU assignments that exceed 100% are common and should not be the sole reason to scale or reject tasks. CPU-based HPA should only be used when the CPU usage goes above 100% for a prolonged period. Instead, memory-based HPA is recommended for autoscaling purposes. Harness suggests using memory-based HPA for better performance and efficiency.
```

> 

Reviewer: ```suggestion
:::
```

       name: harness-delegate
     minReplicas: 2
     maxReplicas: 10
     targetMemoryUtilizationPercentage: 70

   ---
   ```

3. Specify the minimum and maximum number of replicas you want to use in the `minReplicas` and `maxReplicas` parameters.

   To fine-tune your autoscaling, you can set the `targetMemoryUtilizationPercentage` to add a new replica if Memory utilization exceeds this percentage. Below is an example of autoscaling the delegate if the Memory usage of delegates goes above 70%.

4. (Optional) Set the `targetCPUUtilizationPercentage` to add a new replica if CPU utilization exceeds this percentage.

5. Save the file, and restart your pods.

   When you create a deployment, Harness automatically spins up new replicas of your delegate as needed to ensure the deployment is completed.

## Example delegate YAML

> Here's an example YAML file which configures delegates to have a minimum of 2 replicas. If the Memory usage goes above 80%, new tasks won't be accepted. Once the Memory usage hits 70%, a new pod will be created to handle the load, up to a maximum of 10 replicas. When the Memory usage goes back down below 70%, the number of replicas will be scaled back down to a minimum of 2.

Reviewer: ```suggestion
Here's an example YAML file that configures delegates to have a minimum of 2 replicas. New tasks won't be accepted if the Memory usage goes above 80%. Once the Memory usage hits 70%, a new pod will be created to handle the load, up to a maximum of 10 replicas. When the Memory usage goes down below 70%, the number of replicas will be scaled back down to a minimum of 2.
```
